src/main/java/uk/gov/justice/hmpps/prison/util/CalcDateRanges.java

```diff
@@ -68,12 +69,15 @@ public Range<LocalDate> getDateRange() {
     }
 
     public static boolean eventStartsInTimeslot(final LocalDateTime start, final TimeSlot timeSlot) {
-        final var midday = midday(start.toLocalDate());
-        final var evening = evening(start.toLocalDate());
-        return timeSlot == null
-                || (timeSlot == TimeSlot.AM && start.isBefore(midday))
-                || (timeSlot == TimeSlot.PM && !start.isBefore(midday) && start.isBefore(evening))
-                || (timeSlot == TimeSlot.ED && !start.isBefore(evening));
+        return timeSlot == null || eventStartsInTimeslots(start, Set.of(timeSlot));
+    }
+
+    public static boolean eventStartsInTimeslots(final LocalDateTime start, final Set<TimeSlot> timeSlots) {
+        final var hourOfDay = start.getHour();
+        return timeSlots == null || timeSlots.isEmpty()
+            || (timeSlots.contains(TimeSlot.AM) && hourOfDay < 12)
```

Comment:
tried to simplify logic by just looking at the hour of the day rather than playing around with `LocalDateTime`